go/arrow/flight/record_batch_writer.go

```diff
@@ -46,17 +47,39 @@ func (f *flightPayloadWriter) WritePayload(payload ipc.Payload) error {
 
 	payload.SerializeBody(&f.buf)
 	f.fd.DataBody = f.buf.Bytes()
+
 	return f.w.Send(&f.fd)
 }
 
 func (f *flightPayloadWriter) Close() error { return nil }
 
+type Writer struct {
+	*ipc.Writer
+	pw *flightPayloadWriter
+}
+
+// ClearAppMetadata sets the writers app metadata to nil so that subsequent records
+// will not contain App Metadata until another call is made to WriteWithAppMetadata.
+func (w *Writer) ClearAppMetadata() {
+	w.pw.fd.AppMetadata = nil
+}
```

Comment:
Is the idea that you would call `ClearAppMetadata()` after calling `WriteWithAppMetadata()` but before calling `Write()` (with no metadata)?  If so, it might be nice to put this pattern in an integration test, since the tests are probably serving as documentation to most users for now (at least it did to me).

That being said, what do you think about just setting `w.pw.fd.AppMetadata = nil` in `WriteWithAppMetadata()` after `w.Write(rec)`?  So that you don't have to remember calling `ClearAppMetadata()`.

At least the way I've been using it, "sticky" metadata would be an unfortunate bug, and it would be nice to minimize the chances of that.  I suppose my mental model about non-sticky metadata could be wrong, but at least the Java client assumes it is not sticky.

Reply:
fair point. I've updated this without the ClearAppMetadata and made it no longer "sticky". Metadata only sticks around for a single write.